### ISSUE TITLE
algen tests contains xfailing tests that should now pass

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#126]
While reviewing another issue I noticed that there are a couple of tests in `alggen_test.py` that are marked as xfail because they require multiple quadrature objects to be passed in to a single invoke. However, PSyclone does now support that (since #76 I think) and so these tests should be passing.

**Comment by hiker** [comment]
Not without a change - I quickly tried. Note that I refactored that file in #118 (removed the outer class to remove a pylint warning - https://github.com/stfc/PSyclone/pull/122/commits/44561690979d9d217763e7cada3013e271939ee3) - that might create some conflicts if alggen_test is changed before the pull request is merged (assuming that you want my change to alggen_test).

**Comment by arporter** [comment]
Yes, I agree. I only opened this now as a reminder to myself to look into it. I don't intend to do anything about it until I've made in-roads into my reviewing backlog :-)
